### PR TITLE
[profiler] Fix recorded profiler step number

### LINKED ISSUE #121126: The recorded step number in profiler is wrong.
### 🐛 Describe the bug

``` python
def test():
    import torch
    x = torch.randn(10, 10).cuda()
    y = torch.randn(10, 10).cuda()
    with torch.profiler.profile(
            activities=[
                torch.profiler.ProfilerActivity.CPU,
                torch.profiler.ProfilerActivity.CUDA,
            ],
            schedule=torch.profiler.schedule(
                wait=1,
                warmup=2,
                active=3
            ),
            on_trace_ready=torch.profiler.tensorboard_trace_handler('./log'),
        ) as p:
        for _ in range(6):
           z = torch.matmul(x, y)
           p.step()

test()
```
After running above code. I found the first recorded step is Step#2. 
image

According to the scheduling strategy, profiling should wait 1 step and warmup 2 steps, so the first recorded step should be Step#3.

`p.step()` [Signals the profiler that the next profiling step has started.](https://github.com/pytorch/pytorch/blob/main/torch/profiler/profiler.py#L652), so the recorded step number should be next step's.

### Versions

main

cc @robieta @chaekit @aaronenyeshi @nbcsm @guotuofeng @guyang3532 @gaoteng-git @tiffzhaofb @dzhulgakov @davidberard98

### PR DESCRIPTION
**Pull request by fwenguang**
Fixes [121126](https://github.com/pytorch/pytorch/issues/121126)

### CODE REVIEW

**Comment by aaronenyeshi** [comment]
cc @briancoutinho who may be the last person to change this logic, can you please help review?

**Comment by briancoutinho** [comment]
Could you also update the initial step count here from -1 to 0, then everything will be aligned to collect traces based on iteration count
https://github.com/pytorch/pytorch/blob/main/torch/autograd/profiler.py#L996

I think the unit test should work 
https://github.com/pytorch/pytorch/blob/main/test/profiler/test_profiler.py#L1368

**Comment by fwenguang** [comment]
> Could you also update the initial step count here from -1 to 0, then everything will be aligned to collect traces based on iteration count https://github.com/pytorch/pytorch/blob/main/torch/autograd/profiler.py#L996
> 
> I think the unit test should work https://github.com/pytorch/pytorch/blob/main/test/profiler/test_profiler.py#L1368

Okey, updated.

**Review by briancoutinho — APPROVED** [review]
thank you for updates :)
